TACC/Core-Portal
# task/FP-1212 - Use Download Modal for Multiple Files

## Overview: ##

Use Download Modal for multiple files.  Currently, if a user selects multiple files, the download button is disabled. We want to reenable the button, but instead open a download modal like we do for folders. 

## Related Jira tickets: ##

* [FP-1212](https://jira.tacc.utexas.edu/browse/FP-1212)

## Summary of Changes: ##

- Modified `filePermission.js`,  replacing boolean `isFolderSelected` with boolean `areMultipleFilesOrFolderSelected`.  This variable partially controls whether or not the Download button in the toolbar is enabled/disabled.  Now returns true if multiple files, folder(s), or a mix of file(s) and folder(s) are selected for download.  Returns false if one or less files is selected or if `api === googledrive`.
- Modified `DataFilesToolbar.js` to use `areMultipleFilesOrFolderSelected` variable.
- Updated modal wording to include multiple files (wording from Craig Jansen).
- Increased width of modal `size='lg'` (at Craig's request) to prevent wrapping of warning message after wording update.
- Updated `client/src/utils/filePermissions.test.js` to test against `areMultipleFilesOrFolderSelected`.
- Add multiple files fixture and test to `client/src/utils/filePermissions.test.js` to test cases with multiple files.

## Testing Steps: ##
1.  Navigate to 'Data Files' tab in workbench.
2. Select one file and click the 'Download' button from the toolbar.  The file should download.
3. Select two or more files and click the 'Download' button.  The Download (need to compress) modal should be displayed.
4. Select one folder and click the 'Download' button.    The Download (need to compress) modal should be displayed.
5. Select a combination of file(s) and folder(s) and click the 'Download' button.  The Download (need to compress) modal should be displayed.
6. Select Google Drive as the data source.  The Download button should not appear in the toolbar.

## UI Photos:

Downloading a single file.

https://user-images.githubusercontent.com/77903391/135152009-26b3ae3f-9fc6-4ea5-9932-de82b80bae09.mp4

Downloading a folder.  Should get same results for multiple folders, multiple files, or a mix of file(s) and folder(s).

https://user-images.githubusercontent.com/77903391/135154238-fee6d13a-872f-49df-b707-a43643f52360.mp4

## Notes: ##
The related changes from [FP-1102](https://jira.tacc.utexas.edu/browse/FP-1102) are not affected by the changes made in this branch and vice versa.  

At Juan's suggestion, requested and received updated wording from Craig to include multiple file compression.  Craig suggested that modal width be increased to prevent warning message from wrapping.

At Nathan's suggestion, I updated `client/src/utils/filePermissions.test.js` to test against `areMultipleFilesOrFolderSelected` case.

At Sal's suggestion, I added a multiple file fixture and test to `client/src/utils/filePermissions.test.js`.
